build.sbt

```diff
@@ -84,6 +84,26 @@ lazy val `quill-jdbc` =
     )
     .dependsOn(`quill-sql-jvm` % "compile->compile;test->test")
 
+lazy val `quill-jdbc-monix` =
```

Comment:
I wonder if we should have a `quill-monix` module that can be specialized for `quill-jdbc` and others

Reply:
You mean as a place to define interfaces? I think that's a great idea to do that and probably refactor `quill-cassandra` into it via `quill-cassandra-monix`. Any chance we could do it as a seperate PR though?

Reply:
Looking at the code it seems that we could reuse more than just the interfaces. I wonder if we should do it in this PR to avoid having to change the APIs after we release 3.0

Reply:
Most of `AsyncJdbcContext` is very jdbc-specific. Maybe we could extract `prepareParams`, the `Runner` implementations, the result-types, and the `execute...` delegates into a more generic context. I'll try to have a look at this tonight.

Reply:
Do you want to do this ahead of #1259 ?

Reply:
yeah, there's the cassandra integration. I still think it'd be nice to have a generic monix module but I don't feel strongly about it. Feel free to make a decision

Reply:
I moved some stuff out to `quill-monix`. It's a bit thin but should work with the cassandra part.